main.go

```diff
@@ -142,7 +146,8 @@ func binarySearch(blockCount int64, targetTime int64) int64 {
 	cacheMutex.Lock()
 	cacheMap[targetTime] = cacheEntry{height: result, timestamp: time.Now()}
 	cacheMutex.Unlock()
-	return result
+	resultStr := strconv.FormatInt(result, 10)
```

Comment:
good point -I did want to use a different variable than just result tho, just to differentiate from the result variable used in this line. just happened to choose resultStr apparently, or maybe chatGPT did lol. may be a cleaner way to do this in general, not sure. seems to work ok tho, so will go with this for now